### PR TITLE
Split build and test builders for web engine

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This splits the web engine into a build phase and a test phase. This requires a few changes:
* Modify the `generate_builder_json.dart` command to produce two separate json builder files, `linux_web_engine_build.json` and `linux_web_engine_test.json`. The former is left as the existing builder with `release_build: true` (which will run in the merge queue) and the latter is added as a new builder that runs in the second CI phase.
* Change the `copy_artifacts_step.dart` to pull built artifacts from the GCS bucket when on LUCI, instead of relying on the CAS bucket. This is necessary to do the tests separately from the build.
* Move the `felt test --copy-artifacts` command from the build steps to the test steps, since the test steps themselves need to pull from GCS instead of relying on the artifacts being populated by CAS.

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]

File: engine/src/flutter/lib/web_ui/dev/generate_builder_json.dart [hunk 1]
```diff
@@ -69,6 +95,7 @@ class GenerateBuilderJsonCommand extends Command<bool> {
       'gn': <String>[
         '--web',
         '--runtime-mode=release',
+        '--no-rbe',
         '--no-goma',
```
[the remark this answers is not in the document]
Comment (eyebrowsoffire) [reply engine/src/flutter/lib/web_ui/dev/generate_builder_json.dart]: We're supposed to have `--no-rbe`, we can't use rbe with emscripten. Jason made this change recently: https://github.com/flutter/engine/pull/57204

But I realized now he didn't change the generator, and it was overwritten when you regenerated the builder json files in this PR: https://github.com/flutter/flutter/pull/160573

So I'm modifying the generator here, we should definitely pass `--no-rbe`

File: engine/src/flutter/lib/web_ui/dev/steps/copy_artifacts_step.dart [hunk 2]
```diff
@@ -5,19 +5,36 @@
...
+import '../common.dart';
 import '../environment.dart';
 import '../exceptions.dart';
 import '../felt_config.dart';
 import '../pipeline.dart';
 import '../utils.dart';
 
+sealed class ArtifactSource {}
+
+class LocalArtifactSource implements ArtifactSource {
```
[the remark this answers is not in the document]
Comment (eyebrowsoffire) [reply engine/src/flutter/lib/web_ui/dev/steps/copy_artifacts_step.dart:21]: No, this step has never consumed this from CAS artifacts. Previously, the `--copy-artifacts` step was used to *populate* the `web_tests/artifacts` directory in CAS. Now, the artifacts are just uploaded directly to GCS by the build builder and downloaded from GCS by the test builder. CAS is no longer used at all to pass these artifacts around, although it is still used in the test step to pass the compiled unit tests to the test runners (but this is unrelated to the artifacts in this step)

[the remark this answers is not in the document]
Comment (yjbanov) [reply engine/src/flutter/lib/web_ui/dev/steps/copy_artifacts_step.dart:21]: And in presubmit this goes through the presubmit GCS realm or something?

[the remark this answers is not in the document]
Comment (eyebrowsoffire) [reply engine/src/flutter/lib/web_ui/dev/steps/copy_artifacts_step.dart:21]: Yes exactly

**Pull request merged**